mlrun/launcher/local.py

```diff
@@ -69,6 +73,13 @@ def launch(
         notifications: typing.Optional[List[mlrun.model.Notification]] = None,
         returns: Optional[List[Union[str, Dict[str, str]]]] = None,
     ):
+        mlrun.utils.helpers.verify_dict_items_type("Inputs", inputs, [str], [str])
+
+        if runtime.spec.mode and runtime.spec.mode not in run_modes:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f'run mode can only be {",".join(run_modes)}'
+            )
```

Comment:
I think these validations are common to all launchers and should occur after enrichment so that we have a single place where we validate the run

Reply:
Maybe we should have pre_validation as well, so we fail early and can have better explainability, wdyt?